File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/WritePropertyValuesRequest.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
import org.opcfoundation.webserver.types.digitaltwin.ObjectServiceContext;

import java.util.Map;

public class WritePropertyValuesRequest {
    private final String id;
    private final Map<String, Variant> propertyNamesAndValues;
    private final ObjectServiceContext context;

    public WritePropertyValuesRequest(
            ObjectServiceContext context,
            Map<String, Variant> propertyNamesAndValues)
    {
        this.id = context.getObjectId().getId();
        this.propertyNamesAndValues = propertyNamesAndValues;
        this.context = context;
    }

    public String getId() {
        return id;
    }

    public Map<String, Variant> getPropertyNamesAndValues() {
        return propertyNamesAndValues;
    }

    public ObjectServiceContext getContext() {
        return context;
    }
}

File: opcua-webservice-node/source/src/server/service/message/digitialtwin/WritePropertyValuesRequest.ts
import { UaVariant } from "opcua-webapi-ts";
import { ObjectServiceContext } from "../../../types/digitaltwin/ObjectServiceContext";

export class WritePropertyValuesRequest {
    private readonly _id: string;
    private readonly _propertyNamesAndValues: Map<string, UaVariant>;
    private readonly _context: ObjectServiceContext;

    constructor(
        context: ObjectServiceContext,
        propertyNamesAndValues: Map<string, UaVariant>) {
        this._id = context.objectId.id;
        this._propertyNamesAndValues = propertyNamesAndValues;
        this._context = context;
    }

    get id(): string {
        return this._id;
    }

    get propertyNamesAndValues(): Map<string, UaVariant> {
        return this._propertyNamesAndValues;
    }

    get context(): ObjectServiceContext {
        return this._context;
    }
}